src/commands/gw2.ts

```diff
@@ -0,0 +1,165 @@
+import { RhobotCommand } from ".";
+import fetch from "node-fetch";
+import { buildNestedCommand } from "./nestedCommand";
+
+/**
+ * Builds the nested GW 2 command.
+ */
+export function buildCommand({
+  prefix,
+  gw2APIKey,
+  commandEnabled = false,
+}: {
+  prefix: string;
+  gw2APIKey?: string;
+  commandEnabled?: boolean;
+}): RhobotCommand | undefined {
+  if (!commandEnabled) {
+    console.log("[INFO] gw2 command disabled. To enable it please follow instructions in the README (TODO).");
+    return undefined;
+  }
+
+  if (!(gw2APIKey)) {
+    console.error(
+      "[ERROR] Unable to create the gw2 command. " +
+        "API key required. Please check your app-config."
+    );
+    return undefined;
+  }
+
+   const commands = {
+       title: buildTitleCommand(),
+       characters: buildCharactersCommand(gw2APIKey),
+       wallet: buildCurrencyCommand(gw2APIKey),
+    };
+  return buildNestedCommand(prefix, "gw2", "Surface GW2 information.", commands);
+}
+
+function buildTitleCommand() {
+    return {
+        run: (message, parameters) => {
+
+          if (parameters.length !== 1) {
+            message.reply("Expecting one title id");
+            return;
+          }
+
+          if (isNaN(Number(parameters[0]))) {
+            message.reply("Invalid title id expected number");
+            return;
+          }
+
+            getTitle(Number(parameters[0]))
+            .then((text) => message.channel.send(text[0]["name"]))
```

Comment:
Is it really `text`? Looks like an array of results; probably a better name for this.
Also consider destructuring the array.

```javascript
.then(([eplicitNameForThisResult]) => message.channel.send(eplicitNameForThisResult.name))
```

I have similar thoughts for the handlers in the other commands that use `text`.